pkg/epp/common/config/configloader.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/framework"
+	"sigs.k8s.io/yaml"
+)
+
+type Config struct {
+	PluginDefinitions []ConfigPluginDefinition `json:"plugin_definitions"`
+	ProfilePicker     BaseConfigProfilePlugin  `json:"profile_picker"`
+	SchedulerProfiles []ConfigSchedulerProfile `json:"scheduler_profiles"`
+}
+
+type BaseConfigPluginDefinition struct {
+	PluginName string          `json:"plugin_name"`
+	Parameters json.RawMessage `json:"parameters"`
+}
+
+type ConfigPluginDefinition struct {
+	BaseConfigPluginDefinition `json:",inline"`
+	Name                       string `json:"name"`
+}
+
+type ConfigSchedulerProfile struct {
+	Name    string                `json:"name"`
+	Plugins []ConfigProfilePlugin `json:"plugins"`
+}
+
+type BaseConfigProfilePlugin struct {
+	Reference *string                     `json:"reference"`
+	Plugin    *BaseConfigPluginDefinition `json:"plugin"`
+}
+
+type ConfigProfilePlugin struct {
+	BaseConfigProfilePlugin `json:",inline"`
+	Weight                  *int `json:"weight"`
+}
+
+// Load config either from supplied text or from a file
+func LoadConfig(configText []byte, fileName string, log logr.Logger) (*Config, error) {
```

Comment:
Defining it similar to a CRD would allow us to use the decoder pkg from k8s, see this example: https://github.com/kubernetes-sigs/jobset/blob/a7a669590b3483715c193e0af53dbb5f4f066b03/pkg/config/config.go#L19